### PR TITLE
refactor(ts): jobpost and event services

### LINKED ISSUE #196: Bytt ut Redux med useContext
Erstatt all redux funksjonalitet og erstatt den med useContext

### PR DESCRIPTION
**Pull request by olros**
Lint, test and build
## Description

Comments/issues/screenshots:
#243 
closes #196 
- Convert jobpost-service to context and typescript
- Convert EventService to typescript and context
- Removes Redux 🎉 
- Builds on #269 and depends on tihlde/Lepton#194

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] The PR includes a `closes #issueID`
- [ ] The PR includes a picture showing visual changes

## Typescript convert checklist

- [ ] All relative imports has been converted to absolute imports
- [ ] All HOC's have been replaced with Hooks
- [ ] Uses theme.spacing() instead of custom padding
- [ ] Uses theme.breakpoints instead of custom media queries

### CODE REVIEW

File: src/api/api.tsx [hunk 1]
```diff
@@ -1,96 +1,104 @@
...
+  PaginationResponse,
+  Cheatsheet,
+} from 'types/Types';
 
 export default {
   // Events
-  getEventItem: (id: string) => {
-    return new IRequest(RequestMethodType.GET, 'events/'.concat(String(id), '/'), undefined);
+  getEvent: (eventId: number) => {
+    return IFetch<Event>(RequestMethodType.GET, 'events/'.concat(String(eventId), '/'), undefined);
```
Comment (Zenjjim) [inline src/api/api.tsx:24]: backtick!!!

Comment (Zenjjim) [reply src/api/api.tsx:24]: På alle takk!!!

Comment (norgor) [reply src/api/api.tsx:24]: true that

File: src/containers/EventDetails/index.js [hunk 2]
```diff
@@ -77,20 +77,14 @@ function EventDetails(props) {
...
-        setEvent({ ...event });
-      }
-    });
-  };
+  useEffect(() => {
+    if (error) {
+      history.replace(URLS.events);
+    }
+    if (event) {
+      history.replace(URLS.events + id + '/' + urlEncode(event.title) + '/');
```
Comment (Zenjjim) [inline src/containers/EventDetails/index.js:85]: backtick

**Review by Zenjjim — APPROVED** [review]

**Pull request merged**